src/Bicep.Core/Resources/ResourceTypeReference.cs

```diff
@@ -12,7 +12,9 @@ namespace Bicep.Core.Resources
 {
     public class ResourceTypeReference
     {
-        private static readonly Regex ResourceTypePattern = new Regex(@"^(?<namespace>[a-z0-9][a-z0-9\.]*)(/(?<type>[a-z0-9\-]+))+@(?<version>(\d{4}-\d{2}-\d{2})(-(preview|alpha|beta|rc|privatepreview))?$)", RegexOptions.IgnoreCase | RegexOptions.ExplicitCapture | RegexOptions.Compiled | RegexOptions.CultureInvariant);
+        private static readonly Regex ResourceTypePattern = new Regex(@"^(?<namespace>[a-z0-9][a-z0-9\.]*)(/(?<type>[a-z0-9\-]+))+@(?<version>[a-z0-9\-])*?", RegexOptions.IgnoreCase | RegexOptions.ExplicitCapture | RegexOptions.Compiled | RegexOptions.CultureInvariant);
```

Comment:
Worth extracting the regex flags to a field?

Reply:
It's fine either way, but I personally don't do that because the flags are usually specific to a particular regex pattern.